Dockerfile.debian

```diff
@@ -0,0 +1,31 @@
+FROM python:3.7.3-stretch AS build
+ARG DOCKER_COMPOSE_GITSHA=unknown
+ENV DOCKER_COMPOSE_GITSHA=$DOCKER_COMPOSE_GITSHA
+
+RUN apt-get update && apt-get install -y \
+    curl \
+    gcc \
+    git \
+    libc-dev \
+    libgcc-6-dev \
+    locales \
+    make \
+    openssl && \
+    pip install virtualenv==16.2.0 tox==2.1.1
+# FIXME(chris-crone): virtualenv 16.3.0 breaks build, force 16.2.0 until fixed
+
+WORKDIR /code/
```

Comment:
Is there a reason for putting this in a separate `Dockerfile`? Might consider combining them;

```Dockerfile
ARG BUILD_PLATFORM=alpine
ARG DOCKER_VERSION=18.09.4

FROM docker:${DOCKER_VERSION} AS docker-cli
FROM debian:stretch-slim      AS runtime-debian
FROM alpine3.9                AS runtime-alpine

FROM python:3.7.3-alpine3.9   AS build-alpine
RUN apk add --no-cache \
 # etc


FROM python:3.7.3-stretch AS build-debian
ENV BUILD_BOOTLOADER=1
RUN apt-get update && apt-get install -y \
 # etc


# Use either alpine or debian for the build
FROM build-${BUILD_PLATFORM} AS build
WORKDIR /code/

ARG GIT_COMMIT=unknown
ENV DOCKER_COMPOSE_GITSHA=${GIT_COMMIT}
COPY . .
RUN tox --notest
RUN script/build/linux-entrypoint

# Use either alpine or debian for the deploy/runtime
FROM runtime-${BUILD_PLATFORM} AS runtime
ENTRYPOINT ["sh", "/usr/local/bin/docker-compose-entrypoint.sh"]
COPY --from=docker-cli /usr/local/bin/docker          /usr/local/bin/docker
COPY --from=build      /usr/local/bin/docker-compose /usr/local/bin/docker-compose
COPY docker-compose-entrypoint.sh /usr/local/bin/
```


Then;

```bash
EXPORT BUILD_PLATFORM=alpine
EXPORT GIT_COMMIT =$(git rev-parse --verify HEAD) 

docker build \
	-t docker-compose:${BUILD_PLATFORM} \
	--build-arg BUILD_PLATFORM \
	--build-arg GIT_COMMIT \
	.

EXPORT BUILD_PLATFORM=debian
docker build \
	-t docker-compose:${BUILD_PLATFORM} \
	--build-arg BUILD_PLATFORM \
	--build-arg GIT_COMMIT \
	.
```

Reply:
Wow! That's exactly what I was thinking to do, my only question is if that would build both everytime or just the necessary stages for the final target.

Reply:
When using buildkit, only the stages that you need will get built. When using the classic build, everything will be built (all the more reasons to use buildkit, as it's awesome!)